ComSemApp/templates/ComSemApp/student/assessment.html

```diff
@@ -80,6 +100,117 @@ <h2 id="timer-text" class="show-expression"></h2>
+  let reading_check = timer(sentence.length/2, READING_OFFSET);
+  //Once reading timer done this starts
+  reading_check.then(() => {
+    $("#timer-bar").removeClass("bg-info").addClass("bg-danger")
+    //Runs the recording timer
+
+    //starts recording
+    recorder.start();
+    form_expression = problems[curr_problem_index]['id'];
+    let recording_check = timer(sentence.length/2, RECORDING_OFFSET);
+    //Once recording timer done this starts
+    recording_check.then(() => { 
+      // stops recording
+      recorder.stop();
+      // keeps next problem button disabled if no more problems left
+      if (curr_problem_index < problems.length -1 ){
+        toggleEnabledNext();
+      }
+      
+
+      // updates the progress bar
+      updateProgressBar();
+
+      recorder.addEventListener("dataAvailable", (e) => {
+        sendPostRequestForm();
+      });
+      $("#expression-text").html('Problem Complete!');
+    });
+  });
+}
+
+function goToNextProb(){
+  toggleEnabledNext(); //disables next probelm button
+  curr_problem_index = curr_problem_index + 1; // indexes to next probelm
+  var sentence = problems[curr_problem_index]['formulation']; //grabs the formulation
+
+  $("#expression-text").html(sentence); // updates formulation
+  console.log("changed expression to " + sentence);
+  $("#timer-bar").addClass("bg-info").removeClass("bg-danger") //changes color of progress bar back to blue
+  timerPromise(sentence); // starts timer
+}
+
+function sendPostRequestForm(){
+  /*
+    takes current data and send it in a post request to the server (SpeakingPracticeAttemptCreateView)
+  */
+  console.log("save in form called");
+
+  let attemptFormData = new FormData(); // creates new form
+  console.log(form_expression)
+  // form_expression used to ensure correct sentence is passed for the correct problem
+  // if, for whatever reason, the user pressed next problem before the sentence is appended to the form
+  // then the next(and wrong) sentence might get appended to the form
+  // form_expression just holds the correct expression for this form
+  attemptFormData.append( 'expression', form_expression); 
+  attemptFormData.append('student', student);
+  console.log(audioBlob)
+  attemptFormData.append('audio', audioBlob);
+  var csrf_token = '{{ csrf_token }}'; // grabs csrf_token for certification
+  $.ajax({ // send POST request
+        type: "POST",
+        url: $("#save_attempt_url").val(), // links to html form above which specifies the url to recieve request
+        data: attemptFormData,
+        processData: false,
+        contentType: false,
+        headers: { // places token in request so that it is certified by server
+           'X-CSRFToken': csrf_token
+        },
+        success: function(response){
+            cs_notification('success', "Attempt Saved")
+        },
+        error: function(jqXHR, textStatus, errorThrown){
+            cs_ajax_error(jqXHR, textStatus, errorThrown)
+        },
+  });
+
```

Comment:
This blank line looks a little funky. Not a hill I'm willing to die on, but it might make the code more readable if removed